File: Go/symmetric_tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func isSymmetric(root *TreeNode) bool {
    if root == nil {
        return true
    }

    return isSymmetricTrees(root.Left, root.Right)
}

func isSymmetricTrees(l *TreeNode, r *TreeNode) bool {
    if l != nil && r != nil {
        return l.Val == r.Val && isSymmetricTrees(l.Left, r.Right) && isSymmetricTrees(l.Right, r.Left)
    }

    return l == nil && r == nil
}

File: Java/SymmetricTree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
        if(root == null) return true;

        return isSymmetricTrees(root.left, root.right);
    }

    public boolean isSymmetricTrees(TreeNode l, TreeNode r) {
        if(l != null && r != null) {
            return l.val == r.val && isSymmetricTrees(l.left, r.right) 
                    && isSymmetricTrees(l.right, r.left);
        }

        return l == null && r == null;
    }
}